### PR TITLE
Add `lr_scheduler_solver` and  customized `lr_scheduler` classes

### PR DESCRIPTION
**Pull request by lightaime**
Add `lr_scheduler_solver` and customized `lr_scheduler` classes to support getting any learning rate scheduler implemented in pyg and pytorch from its name. Maybe a better location than `utils` like `torch_geometric.optim` in case we develop specific optimization methods for GNNs?

### CODE REVIEW

**Review by rusty1s — APPROVED** [review]
Amazing!
